testUtils/testConnector/mod.ts

```diff
@@ -0,0 +1,20 @@
+type ConnectorSource = (nextPageToken: string) => Promise<string>
+
+export async function testConnector(connectorFunction: ConnectorSource) {
+  const data = [];
+  let response;
+  let nextPageToken = "";
+
+  do {
```

Comment:
This looks good thanks! 

When I use this locally I often add logs to see different details about what's happening, for example some times I want to see the actual data being fetched or inspect the page token to make sure I'm constructing them correctly. For people who are importing this function they won't be able to do that, so I think it could be helpful to add some options for changing what gets logged. 

To be as flexible as possible that could maybe be something like an optional second object parameter with some different properties, one could be a callback that gets passed the responses so that the user can do the console logging of response.data.length or response.nextPageToken themselves instead of that being just default. 

Similarly, instead of logging all the data at the end maybe we return an object that contains information like the data or number of requests made so the user can choose what's relevant for them to look at.